backend/src/test/java/capstone/myfinancemanager/manager/controller/TransactionControllerIntegrationTest.java

```diff
@@ -128,4 +131,91 @@ void deleteTransactionNotExists() throws Exception {
+        String id = saveResultTransaction.getId();
+
+        TransactionCreationDto transactionCreationDto = TransactionCreationDto.builder()
+                .userEmail(saveResultTransaction.getUserEmail())
+                .description("Heute ist sehr schön")
+                .amount(159)
+                .transactionDate(Long.parseLong("1661866382913"))
+                .category("Tanken")
+                .pictureId("url2")
+                .isIncome(false)
+                .build();
+
+        String updatedResult = mockMvc.perform(
+                        MockMvcRequestBuilders.put("/transactions/update/" + saveResultTransaction.getId())
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(objectMapper.writeValueAsString(transactionCreationDto))
+                )
+                .andExpect(status().is(201))
+                .andReturn().getResponse().getContentAsString();
+
+        Transaction actualPlant = objectMapper.readValue(updatedResult, Transaction.class);
+        Assertions.assertEquals(saveResultTransaction.getId(), actualPlant.getId());
+    }
+
+
+    @Test
+    @DirtiesContext
+    void updateTransactionDoNotExistTest() throws Exception {
+        String saveResult = mockMvc.perform(post(
+                "/transactions")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content("""
+                           {
+                                    "userEmail": "a@a.com",
+                                    "description": "5",
+                                    "amount": 123.2,
+                                    "transactionDate": "1661866382913",
+                                    "category": "Essen",
+                                    "pictureId": "url",
+                                    "isIncome": true
+                                }
+                        """)
+        ).andReturn().getResponse().getContentAsString(StandardCharsets.UTF_8);
+
+        Transaction saveResultTransaction = objectMapper.readValue(saveResult, Transaction.class);
+        String id = saveResultTransaction.getId();
+
+        TransactionCreationDto transactionCreationDto = TransactionCreationDto.builder()
+                .userEmail(saveResultTransaction.getUserEmail())
+                .description("Heute ist sehr schön")
+                .amount(159)
+                .transactionDate(Long.parseLong("1661866382913"))
+                .category("Tanken")
+                .pictureId("url2")
+                .isIncome(false)
+                .build();
+
+        mockMvc.perform(
+                        MockMvcRequestBuilders.put("/transactions/update/" + "1")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(objectMapper.writeValueAsString(transactionCreationDto))
+                )
+                .andExpect(status().is(403));
```

Comment:
404, see controller

having a test for missing privilege would have revealed commented out code in controller advice ;)